File: back-end/src/main/java/server/model/DokumentiPredmeta.java
package server.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DokumentiPredmeta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String silabus;
    private String akreditacija;

    @OneToOne(mappedBy = "dokumentiPredmeta")
    private Predmet predmet;

    @Column(nullable = false)
    private Boolean vidljiv = true;
}

File: front-end/src/app/models/DokumentiPredmeta.ts
import { Predmet } from "./Predmet";
import { Silabus } from "./Silabus";

export interface DokumentiPredmeta {
  id: number;
  vidljiv: boolean;
  akreditacija: string;
  silabus: Silabus;
  predmet: Predmet;
}
